dotnet-template-samples/content/17-rename-file/MyProject.Con/.template.config/template.json

```
@@ -0,0 +1,26 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Contoso",
+  "classifications": [ "Console" ], 
+  "name": "Contoso Sample 17",
+  "identity": "MyProject.17.Sample.CSharp",         
+  "groupIdentity":"MyProject.17.Sample",
+  "shortName": "sample17",                    
+  "tags": {
+    "language": "C#",                         
+    "type":"project"
+  },
+  "symbols": {
+        "renameFileParameter": {
+            "type": "parameter",
+            "displayName": "File name",
+            "description": "Renames the default FileToRename.cs to the value provided.",
+            "datatype": "text",
+            "defaultValue": "FileToRename",            
+            "replaces": "FileToRename",                // Optionally renames the text inside all files matching 'FileToRename'
+            "fileRename": "FileToRename"               // Finds files and folders including the text 'FileToRename' and replaces it with the value stored in the 'renameFileParameter'
+        }
+      },
+  "sourceName": "MyProject.Con",              
+  "preferNameDirectory":true
```

JSON doesn't officially support comments. Please remove them. Also, here's some whitespace cleanup.

```suggestion
  "$schema": "http://json.schemastore.org/template",
  "author": "Contoso",
  "classifications": [ "Console" ],
  "name": "Contoso Sample 17",
  "identity": "MyProject.17.Sample.CSharp",
  "groupIdentity": "MyProject.17.Sample",
  "shortName": "sample17",
  "tags": {
    "language": "C#",
    "type": "project"
  },
  "symbols": {
    "renameFileParameter": {
      "type": "parameter",
      "displayName": "File name",
      "description": "Renames the default FileToRename.cs to the value provided.",
      "datatype": "text",
      "defaultValue": "FileToRename",
      "replaces": "FileToRename",
      "fileRename": "FileToRename"
    }
  },
  "sourceName": "MyProject.Con",
  "preferNameDirectory": true
```

---

No problem, I just removed and formatted the template.json, comments we included based on the 01-basic-template example [comments](https://github.com/dotnet/templating/blob/main/dotnet-template-samples/content/01-basic-template/MyProject.Con/.template.config/template.json)

---

Oh, that's interesting. 🤔 I wouldn't use that as a good example. 😅

---

I thought it was odd

---

That was probably my fault 😁